runtime/download.go

```diff
@@ -0,0 +1,120 @@
+package runtime
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/user"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/yourbase/yb/plumbing/log"
+)
+
+const defaultCacheDir = "/tmp/yourbase/cache"
+
+func localCacheDir() string {
+	if cacheDir, exists := os.LookupEnv("YB_CACHE_DIR"); exists {
+		return cacheDir
+	}
+	// Resolve current homeDir
+	if u, err := user.Current(); err == nil && u.HomeDir != "" {
+		return filepath.Join(u.HomeDir, ".cache", "yourbase")
+	}
+	// Try again
+	if homeDir, exists := os.LookupEnv("HOME"); exists {
+		return filepath.Join(homeDir, ".cache", "yourbase")
+	}
+	return defaultCacheDir
```

Comment:
In the spirit of [ch1401](https://app.clubhouse.io/yourbaseio/story/1401), let's use [xdgdir](https://pkg.go.dev/go4.org/xdgdir) to find the cache directory.

```suggestion
	if cacheDir := xdgdir.Cache.Path(); cacheDir != "" {
		return filepath.Join(cacheDir, "yourbase")
	}
	return defaultCacheDir
```